### PR TITLE
[GUILD] - Convert Pagination component to typescript

### PR DESCRIPTION
**Pull request by eniskraasniqi1**
### Description

- This PR converts Pagination component to typescript.

**Note**: For Pagination.tsx from line 57 -> Changed Ellipsis check from strings to checking with array indexes. **Reason**: Type confusion on the interface and also props.

### Manual check:
- Component works as before
- Storybook works as before

### CODE REVIEW

File: src/components/pagination/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import React, { ReactNode } from 'react';
+import Box from '../box';
+import { TextBody } from '../typography';
+import { IconChevronLeftMediumOutline, IconChevronRightMediumOutline } from '@teamleader/ui-icons';
+import cx from 'classnames';
+import theme from './theme.css';
+import { BoxProps } from '../box/Box';
+import { GenericComponent } from '../../@types/types';
+
+type ChildrenFunctionArgument = {
```
Comment (qubis741) [inline src/components/pagination/Pagination.tsx:10]: interface

File: src/components/pagination/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  isActive: boolean;
+  icon?: ReactNode;
+  iconPlacement?: 'left' | 'right';
+  text?: string;
+  className?: string;
+};
+
+interface PaginationProps extends Omit<BoxProps, 'className' | 'children'> {
+  className?: string;
+  currentPage: number | undefined;
```
Comment (qubis741) [inline src/components/pagination/Pagination.tsx:21]: use `?` instead of `| undefined`

File: src/components/pagination/Pagination.tsx (hunk 1 above)
Comment (farazatarodi) [reply src/components/pagination/Pagination.tsx:10]: Rule of thumb: Always use `interface` unless you need `type`.

Comment (eniskraasniqi1) [reply src/components/pagination/Pagination.tsx:10]: Yeah, already read this, but don't know how I missed this. Thank you guys!

**Comment by farazatarodi** [comment]
Tip for next time: Separate the renaming of your component and adding changes to it into two commits. This way we keep the file history and the changes are shown on the GitHub comparison view.

**Review by qubis741 — APPROVED** [review]

**Review by BeirlaenAaron — APPROVED** [review]

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

**Pull request merged**